javascript-challenges.test.js

```diff
@@ -21,9 +23,35 @@ const secretCodeWord2 = "Gobbledygook"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
+describe("strings", () =>{
+    it("takes in a string and returns a string with a coded message The coded message converts a to 4, e to 3, i to 1, and o to 0", => {
+    expect((secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+    expect((secretCodeWord2)).toEqual("G0bbl3dyg00k")
+    expect((secretCodeWord3)).toEqual("3cc3ntr1c")
+})
+})
 // b) Create the function that makes the test pass.
-
-// Pseudo code:
+const secretCode = (inputString) => {
+    const letterMap = {
+        'a': '4',
+        'e': '3',
+        'i': '1',
+        'o': '0',
+    }
+    const codedMessageArray = inputString.split('').map(letter => {
+        return wordMap[letter] || letter
+    })
+    return codedMessageArray.join('')
+}
+console.log(secretCode(secretCodeWord1))
+console.log(secretCode(secretCodeWord2))
+console.log(secretCode(secretCodeWord3))
+// // Pseudo code:
+// 1. created a function that takes on an string as a arguemnt 
+// 2. split input string into array and use .map for each letter or number
+// 3. wordMap is used to to find if the letter is not in the map it will return the letter back
+// 4. using the join method we join the array back into a string 
+// 5. (i UNderstand how this code works bu this is a nother one i got a lot of help on and im not sure if i could explain it myself)
```

Comment:
I know what you're trying to do here, but it actually doesn't work.  The biggest problem here is that I can tell you didn't run your tests.  The test fails.  Please base your function code on your test, not on the console.log result.  Keep pushing, Chandler.  I see a lot of improvement!